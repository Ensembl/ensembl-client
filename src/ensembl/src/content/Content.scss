@import 'src/styles/common';

/*
 TODO:
  Add a global layout component based on CSS grid:

  | HEADER  |
  | CONTENT |

  so that the area allotted to content is calculated automatically by the browser,
  and not specifically defined as currently is done below
*/

.content {
  &.shorter {
    height: calc(100% - 79px);
  }

  &.taller {
    height: calc(100% - 25px);
  }
}
