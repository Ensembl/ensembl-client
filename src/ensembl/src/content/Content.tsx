import React, { FunctionComponent } from 'react';
import { Route, withRouter } from 'react-router-dom';
import { RouteComponentProps } from 'react-router';
import { connect } from 'react-redux';

import { RootState } from '../rootReducer';
import { getLaunchbarExpanded } from '../header/headerSelectors';

import Home from './home/Home';
import App from './app/App';

import styles from './Content.scss';

type StateProps = {
  launchbarExpanded: boolean;
};

type OwnProps = {};

type ContentProps = RouteComponentProps & StateProps & OwnProps;

export const getHeightClass = (launchbarExpanded: boolean): string => {
  return launchbarExpanded ? styles.shorter : styles.taller;
};

export const Content: FunctionComponent<ContentProps> = (
  props: ContentProps
) => {
  return (
    <main
      className={`${styles.content} ${getHeightClass(props.launchbarExpanded)}`}
    >
      <Route path="/" component={Home} exact={true} />
      <Route path="/app" component={App} />
    </main>
  );
};

const mapStateToProps = (state: RootState): StateProps => ({
  launchbarExpanded: getLaunchbarExpanded(state)
});

export default withRouter(connect(mapStateToProps)(Content));
