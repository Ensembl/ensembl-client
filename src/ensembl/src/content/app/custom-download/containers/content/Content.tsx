import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import styles from './Content.scss';
import {
  getSelectedTabButton,
  getShowPreviewResult,
  getPreviewResult
} from '../../customDownloadSelectors';
import { setAttributes } from './attributes-accordion/attributesAccordionActions';
import { getAttributes } from './attributes-accordion/attributesAccordionSelector';

import AttributesAccordion from './attributes-accordion/AttributesAccordion';
import FiltersAccordion from './filter-accordion/FiltersAccordion';
import TabButtons from './tab-buttons/TabButtons';
import ResultHolder from './result-holder/ResultHolder';
import { RootState } from 'src/store';
import { attributes } from '../../sampledata';
import PreviewTable from '../../components/preview-table/PreviewTable';
// import { fetchGeneAttributes } from 'src/services/custom-download';

import { getSelectedAttributes, formatResults } from './result-holder/helpers';

type Props = StateProps & DispatchProps;

// const getGeneAttributes = async (props: Props) => {
//   const allAttributes = await fetchGeneAttributes();

//   const attributes: any = {};

//   let sections = ['gene'];

//   allAttributes.forEach((attribute: any) => {
//     sections = ['gene'];
//     if (attribute.type === 'TERM') {
//       if (attribute.name.split('.').length > 1) {
//         sections = attribute.name.split('.');
//       }
//     }

//     if (!attributes[sections[0]]) {
//       attributes[sections[0]] = {};
//     }
//     if (sections[0] && sections.length !== 1 && !attributes[sections[0]]) {
//       attributes[sections[0]] = {};
//     } else if (sections.length == 1) {
//       if(!attributes[sections[0]]['default']){
//         attributes[sections[0]]['default'] = {};
//       }
//       attributes[sections[0]]['default'][attribute.name] = {
//         id: attribute.name,
//         label: attribute.displayName,
//         checkedStatus: false
//       };
//     }

//     if (
//       sections[1] &&
//       sections.length !== 2 &&
//       !attributes[sections[0]][sections[1]]
//     ) {
//       attributes[sections[0]][sections[1]] = {};
//     } else if (sections.length == 2) {
//       if(!attributes[sections[0]]['default']){
//         attributes[sections[0]]['default'] = {};
//       }
//       attributes[sections[0]]['default'][sections[1]] = {
//         id: attribute.name,
//         label: attribute.displayName,
//         checkedStatus: false
//       };
//     }

//     if (
//       sections[2] &&
//       sections.length !== 3 &&
//       !attributes[sections[0]][sections[1]][sections[2]]
//     ) {
//       attributes[sections[0]][sections[1]][sections[2]] = {};
//     } else if (sections.length == 3) {
//       if(!attributes[sections[0]][sections[1]]['default']){
//         attributes[sections[0]][sections[1]]['default'] = {};
//       }
//       attributes[sections[0]][sections[1]]['default'][sections[2]] = {
//         id: attribute.name,
//         label: attribute.displayName,
//         checkedStatus: false
//       };
//     }

//     if (
//       sections[3] &&
//       sections.length !== 4 &&
//       !attributes[sections[0]][sections[1]][sections[2]][sections[3]]
//     ) {
//       attributes[sections[0]][sections[1]][sections[2]][sections[3]] = {};
//     } else if (sections.length == 4) {

//       if(!attributes[sections[0]][sections[1]][sections[2]]['default']){
//         attributes[sections[0]][sections[1]][sections[2]]['default'] = {};
//       }

//       attributes[sections[0]][sections[1]][sections[2]]['default'][sections[3]] = {
//         id: attribute.name,
//         label: attribute.displayName,
//         checkedStatus: false
//       };
//     }
//   });
//   props.setAttributes(attributes);
// };

const Content = (props: Props) => {
  useEffect(() => {
    // getGeneAttributes(props);
    props.setAttributes(attributes);
  }, []);

  const selectedAttributes: any = getSelectedAttributes(props.attributes);
  let formatedPreviewResult = [];
  if (props.previewResult.results) {
    formatedPreviewResult = formatResults(
      props.previewResult,
      selectedAttributes
    );
  }

  return (
    <div>
      {!props.showPreview && (
        <div>
          <div className={styles.resultList}>
            <ResultHolder />
          </div>
          <div className={styles.tabList}>
            <TabButtons />
          </div>
          <div className={styles.dataSelector}>
            {props.selectedTabButton === 'attributes' && (
              <AttributesAccordion />
            )}
            {props.selectedTabButton === 'filter' && <FiltersAccordion />}
          </div>
        </div>
      )}
      {props.showPreview && (
        <div>
          {' '}
          <PreviewTable resultData={formatedPreviewResult} />{' '}
        </div>
      )}
    </div>
  );
};

type StateProps = {
  selectedTabButton: string;
  showPreview: boolean;
  previewResult: any;
  attributes: any;
};

const mapStateToProps = (state: RootState): StateProps => ({
  selectedTabButton: getSelectedTabButton(state),
  showPreview: getShowPreviewResult(state),
  previewResult: getPreviewResult(state),
  attributes: getAttributes(state)
});

type DispatchProps = {
  setAttributes: (setAttributes: {}) => void;
};

const mapDispatchToProps: DispatchProps = {
  setAttributes
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Content);
