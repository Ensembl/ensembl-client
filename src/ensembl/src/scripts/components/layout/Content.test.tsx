import React from 'react';
import {
  withRouter,
  RouteComponentProps,
  MemoryRouter
} from 'react-router-dom';
import { mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Content } from './Content';

describe('<Content />', () => {
  let WrappedComponent: any;

  beforeEach(() => {
    WrappedComponent = withRouter((props: RouteComponentProps) => (
      <Content launchbarExpanded={true} {...props} />
    ));
  });

  describe('<main> element', () => {
    let wrapper: any;

    beforeEach(() => {});

    test('collapses correctly', () => {
      wrapper = mount(
        <MemoryRouter>
          <WrappedComponent />
        </MemoryRouter>
      );

      expect(
        wrapper
          .find(Content)
          .instance()
          .getExpandClass()
      ).toBe('');
    });

    test('expands correctly', () => {
      WrappedComponent = withRouter((props: RouteComponentProps) => (
        <Content launchbarExpanded={false} {...props} />
      ));

      wrapper = mount(
        <MemoryRouter>
          <WrappedComponent />
        </MemoryRouter>
      );

      expect(
        wrapper
          .find(Content)
          .instance()
          .getExpandClass()
      ).toBe('expanded');
    });
  });

  test('renders correctly', () => {
    const wrapper = render(
      <MemoryRouter>
        <WrappedComponent />
      </MemoryRouter>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
