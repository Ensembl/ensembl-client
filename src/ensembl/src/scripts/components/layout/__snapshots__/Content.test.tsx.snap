// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> renders correctly 1`] = `
<main
  class=""
/>
`;
